modules/runners/templates/user-data.sh

```diff
@@ -7,13 +7,20 @@ yum update -y
 
 # Install docker
 amazon-linux-extras install docker
+# Installing helper to login to ECR using EC2 IAM role
+yum install -y amazon-ecr-credential-helper
 service docker start
 usermod -a -G docker ec2-user
 
 yum install -y curl jq git
 
 # Install runner
 cd /home/ec2-user
+# Adding credHelper to login to ECR using EC2 IAM role
+mkdir .docker && echo "{\"credHelpers\": {\"604664627747.dkr.ecr.eu-west-1.amazonaws.com\": \"ecr-login\"}}" >> .docker/config.json
```

Comment:
Doesn't module provide a place to override userdata. This should ideally be present outside the module as modules are generic and this is too specific to what you are trying to do